src/exchange.rs

```diff
@@ -22,6 +23,20 @@ impl<'a, T: FormattableCurrency> Exchange<'a, T> {
         self.map.insert(key, *rate);
     }
 
+    pub fn set_rate_and_inverse(&mut self, rate: &'a ExchangeRate<T>) -> Result<ExchangeRate<'a, T>, MoneyError> {
```

Comment:
Return a result with inverse rate or `MoneyError` since it would be weird to return `Option<MoneyError>` which was the other option I considered